eduHub/src/main/kotlin/com/arsiu/eduhub/controller/nats/assignment/AssignmentNatsControllerUpdate.kt

```diff
@@ -0,0 +1,72 @@
+package com.arsiu.eduhub.controller.nats.assignment
+
+import com.arsiu.eduhub.controller.nats.NatsController
+import com.arsiu.eduhub.mapper.AssignmentNatsMapper
+import com.arsiu.eduhub.service.AssignmentService
+import com.arsiu.eduhub.v2.assignmentsvc.NatsSubject.ASSIGNMENT_UPDATE_BY_ID
+import com.arsiu.eduhub.v2.assignmentsvc.commonmodels.assignment.AssignmentProto
+import com.arsiu.eduhub.v2.assignmentsvc.commonmodels.assignment.AssignmentResponse
+import com.arsiu.eduhub.v2.assignmentsvc.commonmodels.assignment.AssignmentResponse.Failure
+import com.arsiu.eduhub.v2.assignmentsvc.commonmodels.assignment.AssignmentResponse.Failure.CustomError
+import com.arsiu.eduhub.v2.assignmentsvc.commonmodels.assignment.AssignmentResponse.Success
+import com.arsiu.eduhub.v2.assignmentsvc.input.reqreply.assignment.UpdateAssignmentRequest
+import com.arsiu.eduhub.v2.assignmentsvc.output.reqreply.assignment.UpdateAssignmentResponse
+import com.google.protobuf.Parser
+import io.nats.client.Connection
+import org.springframework.stereotype.Component
+
+@Component
+class AssignmentNatsControllerUpdate(
+    override val connection: Connection,
+    private val service: AssignmentService,
+    private val mapper: AssignmentNatsMapper
+) : NatsController<UpdateAssignmentRequest, UpdateAssignmentResponse> {
+
+    override val subject: String = ASSIGNMENT_UPDATE_BY_ID
+    override val parser: Parser<UpdateAssignmentRequest> = UpdateAssignmentRequest.parser()
+
+    override fun handler(request: UpdateAssignmentRequest): UpdateAssignmentResponse {
+        return runCatching {
+            val obj = mapper.toEntityUpdate(request.request.assignment)
+            val updated = service.update(obj)
+            getSuccessResponse(mapper.toResponseDto(updated))
+        }.getOrElse { ex ->
+            getFailureResponse(ex.javaClass.simpleName, ex.toString())
+        }
+    }
+
+    override fun getSuccessResponse(obj: Any): UpdateAssignmentResponse {
+        val successResponse = Success.newBuilder()
+            .setMessage("Assignment updated successfully")
+            .setAssignment(obj as AssignmentProto)
+            .build()
+
+        val response = AssignmentResponse.newBuilder()
+            .setSuccess(successResponse)
+            .build()
+
+        return UpdateAssignmentResponse.newBuilder()
+            .setResponse(response)
+            .build()
```

Comment:
```suggestion
        return UpdateAssignmentResponse.newBuilder().apply {
            responseBuilder.successBuilder
                    .setMessage("Assignment updated successfully")
                    .setAssignment(obj as AssignmentProto)
        }.build()
```

What about this approach? :)

Reply:
the same trick could be applied to failure response builder

Reply:
`+`